CSharpSourceCode/Battle/StatusEffects/StatusEffectMissionLogic.cs

```diff
@@ -15,28 +16,59 @@ namespace TOW_Core.Battle.StatusEffects
 {
     public class StatusEffectMissionLogic : MissionLogic
     {
+        private Dictionary<int, Agent> _activeAgents;
+        private bool init;
+
+      
         public override void OnAgentCreated(Agent agent)
         {
+            if (_activeAgents == null&&!init)
+            {
+                _activeAgents = new Dictionary<int, Agent>(); 
+                init = true;
+            }
+            
             if (agent.IsHuman)
             {
                 StatusEffectComponent effectComponent = new StatusEffectComponent(agent);
                 agent.AddComponent(effectComponent);
+                _activeAgents?.Add(agent.Index, agent);
             }
         }
         
         public override void OnMissionTick(float dt)
         {
-            foreach(var agent in Mission.Current.Agents)
+            
+            if (!init)
+                return;
+
+            foreach (var agent in _activeAgents.ToDictionary(x =>x.Key, x=> x.Value))
```

Comment:
_activeAgents is Dictionary already

Reply:
this is needed to be able to change the dictionary within the for loop